xbmc/epg/EpgContainer.cpp

```diff
@@ -65,7 +65,10 @@ CEpgContainer::CEpgContainer(void) :
 
 CEpgContainer::~CEpgContainer(void)
 {
-  Unload();
+  // Don't call Unload here, it'll try
+  // to notify observers which may be in
+  // different states of teardown
+  Stop();
```

Comment:
Unload() does more stuff but when looking at it, none of it seemed important in a destructor as all objects will clean themselves up. I may be mistaken but it looked perfectly safe to remove the call to it.